### PR TITLE
[feat] Update mpt and poseidon circuit with new hash scheme

### PR DESCRIPTION
**Pull request by noel2004**
This PR has integrated mutiple new dependencies for updating the new hash scheme applied in zktrie:

zktrie: integrated new `v0.6` branch
- [x] FFI test passed
- [x] `estimate row` test in scroll-prover passed, indicating the new trace can be normal parsed and handled

mpt-circuit: pending for update

poseidon-circuit: integrated new `scroll0-dev-0723` branch
- [x] poseidon codehash related update passed unittests

mock prove passed with `zktrie` feature and two `mpt_update exists` lookup in state circuit being disabled

### CODE REVIEW

File: zkevm-circuits/src/table.rs [hunk 1]
```diff
@@ -948,26 +953,26 @@ impl PoseidonTable {
...
-                    .zip(once(nil_hash).chain(repeat(Value::known(F::zero()))))
-                {
-                    region.assign_advice(
-                        || "poseidon table nil input row",
-                        column,
-                        offset,
-                        || value,
-                    )?;
-                }
+                // let nil_hash =
```
Comment (z2trillion) [inline zkevm-circuits/src/table.rs:956]: is this ok to delete?

File: zktrie/src/state/witness.rs [hunk 2]
```diff
@@ -501,6 +503,7 @@ fn decode_proof_for_mpt_path(mut key: Word, proofs: Vec<Vec<u8>>) -> Result<SMTP
     }
 
     let leaf = trie_proof.key.as_ref().map(|h| SMTNode {
+        node_type: 4,
```
Comment (z2trillion) [inline zktrie/src/state/witness.rs]: can you use `NODE_TYPE_LEAF` instead of 4 here?

**Comment by noel2004** [comment]
we have passed mock prove with new erc20, sushi, nft and unitswap trace

**Review by z2trillion — APPROVED** [review]
looks great. I just have two small requested changes.

File: zktrie/src/state/witness.rs (hunk 2 above)
Comment (noel2004) [reply zktrie/src/state/witness.rs]: We do not export the constant defination of node type from FFI and it maybe error-prone to make duplicated defination here. However, the node type can be always obtained while parsing the trie data so i would update it by this way.

**Comment by lispc** [comment]
hold. wait for l2geth merging zktrie 0.6

**Pull request merged**